dashboards/support.jsonnet

```diff
@@ -8,210 +8,281 @@ local row = grafonnet.panel.row;
       |||
     )
-    + prometheus.withLegendFormat('{{ device }}'),
+    + prometheus.withLegendFormat('receive ({{namespace}})'),
+    prometheus.new(
+      '$PROMETHEUS_DS',
+      |||
+        sum(irate(container_network_transmit_bytes_total{pod=~".*-home-nfs-.*"}[5m])) by (namespace)
+      |||
+    )
+    + prometheus.withLegendFormat('transmit ({{namespace}})'),
   ]);
 
-local nfsServerReadLatency =
+
+// Prometheus server diagnostics
+// -----------------------------
+local promServerCPU =
   common.tsOptions
-  + ts.new('NFS Server disk read latency')
+  + common.tsRequestLimitStylingOverrides
+  + ts.new('Prometheus server CPU usage')
+  + ts.standardOptions.withUnit('sishort')
   + ts.queryOptions.withTargets([
     prometheus.new(
       '$PROMETHEUS_DS',
       |||
-        sum(rate(node_disk_read_time_seconds_total{job="prometheus-nfsd-server"}[5m])) by (device) / sum(rate(node_disk_reads_completed_total{job="prometheus-nfsd-server"}[5m])) by (device)
+        sum(irate(container_cpu_usage_seconds_total{pod=~".*prometheus-server-.*", container!=""}[5m])) by (namespace, container)
       |||
     )
-    + prometheus.withLegendFormat('{{ device }}'),
+    + prometheus.withLegendFormat('{{namespace}}: {{container}}'),
+    prometheus.new(
+      '$PROMETHEUS_DS',
+      |||
+        sum(kube_pod_container_resource_requests{pod=~".*prometheus-server-.*", container!="", resource="cpu"}) by (namespace, container)
+      |||
+    )
+    + prometheus.withLegendFormat('request ({{namespace}}: {{container}})'),
+    prometheus.new(
+      '$PROMETHEUS_DS',
+      |||
+        sum(kube_pod_container_resource_limits{pod=~".*prometheus-server-.*", container!="", resource="cpu"}) by (namespace, container)
+      |||
+    )
+    + prometheus.withLegendFormat('limit ({{namespace}}: {{container}})')
+    + prometheus.withHide(),
   ]);
 
-// Support Metrics
-
-// FIXME: Can we transition to using the function to generate the prometheus memory and cpu panels?
-//
-//        Currently held back by hardcoded label selection on the label
-//        "component" and selection on a single label instead of optionally
-//        multiple.
-//
-//local prometheusMemory = jupyterhub.memoryPanel(
-//  'Prometheus Memory (Working Set)',
-//  // app.kubernetes.io/component: server
-//  // app.kubernetes.io/name: prometheus
-//  component='singleuser-server',
-//  multi=false,
-//);
-
-local prometheusMemory =
+local promServerMemory =
   common.tsOptions
-  + ts.new('Prometheus Memory (Working Set)')
+  + common.tsRequestLimitStylingOverrides
+  + ts.new('Prometheus server memory usage (working set)')
   + ts.standardOptions.withUnit('bytes')
   + ts.queryOptions.withTargets([
     prometheus.new(
       '$PROMETHEUS_DS',
       |||
-        sum(container_memory_working_set_bytes{pod=~"support-prometheus-server-.*", namespace="support"})
+        sum(container_memory_working_set_bytes{pod=~".*prometheus-server-.*", container!=""}) by (namespace, container)
```

Comment:
Thanks for removing some of these namespaces in the matching strings, and adding `container!=""`